src/mlang/optimizing_ir/partial_evaluation.ml

```diff
@@ -618,27 +618,13 @@ let rec partially_evaluate_expr (ctx : partial_ev_ctx) (p : Mir.program)
             from_literal (Bir_interpreter.evaluate_expr p new_e RegularFloat)
           else
             match func with
-            | ArrFunc | InfFunc -> (Pos.unmark new_e, List.hd new_ds)
+            | ArrFunc | InfFunc | MinFunc | MaxFunc | Multimax ->
+                (Pos.unmark new_e, List.hd new_ds)
             | PresentFunc -> (
                 match List.hd new_ds with
                 | Undefined -> from_literal Mir.false_literal
                 | Float -> from_literal Mir.true_literal
                 | _ -> (Pos.unmark new_e, Float))
-            | MinFunc | MaxFunc | Multimax ->
-                (* in the functions, undef is implicitly cast to 0, so let's
-                   cast it! *)
-                let new_args =
-                  List.map2
-                    (fun a d ->
-                      if Pos.unmark a = Mir.Literal Undefined || d = Undefined
-                      then Pos.same_pos_as (Mir.Literal (Float 0.)) a
-                      else a)
-                    new_args new_ds
-                in
-                let new_e =
-                  Pos.same_pos_as (Mir.FunctionCall (func, new_args)) e
-                in
-                (Pos.unmark new_e, Float)
```

Comment:
It's a shame not to reduce those when we can.

Please comment that we should do it at some point.

Reply:
Ah, too late. Well then, later I guess.

Reply:
Would you like an issue on the topic ?

Reply:
#194 :)

Reply:
To be clearer, it does not patch multimax. But I don't think it's worth trying to considering how the pass works and how little chance there is to actually simplify it statically.